features/steps/thens/attributes.py

```diff
@@ -60,37 +60,124 @@ def accumulate_errors(i):
         yield from errors
 
 
-@gherkin_ifc.step('The value of attribute {attribute} must be {value}')
-@gherkin_ifc.step('The value of attribute {attribute} must be {value} {display_entity:display_entity}')
-def step_impl(context, inst, attribute, value, display_entity=0):
-    # @todo the horror and inconsistency.. should we use
-    # ast here as well to differentiate between types?
-    pred = operator.eq
-    if value == 'empty':
-        value = ()
-    elif value == 'not empty':
-        value = ()
-        pred = operator.ne
-    elif 'or' in value:
-        opts = value.split(' or ')
-        value = tuple(opts)
-        pred = misc.reverse_operands(operator.contains)
-
-    if isinstance(inst, (tuple, list)):
-        inst = inst[0]
-    attribute_value = getattr(inst, attribute, 'Attribute not found')
-    if attribute_value is None:
-        attribute_value = ()
-    if inst is None:
-        # nothing was activated by the Given criteria
-        yield ValidationOutcome(inst=inst, severity=OutcomeSeverity.EXECUTED)
-    elif not pred(attribute_value, value):
-        yield ValidationOutcome(
-            inst=inst,
-            expected=None if not value else value,
-            observed=misc.recursive_unpack_value(attribute_value),
-            severity=OutcomeSeverity.ERROR
-        )
+@gherkin_ifc.step('The value of attribute {attribute} must be {value_or_comparison_op}')
+@gherkin_ifc.step('The value of attribute {attribute} must be {value_or_comparison_op} {display_entity:display_entity}')
+@gherkin_ifc.step('The value of attribute {attribute} must be {value_or_comparison_op} the expression: {expression}')
+def step_impl(context, inst, attribute:str, value_or_comparison_op:str, expression:str=None, display_entity=0):
+    """
+    Compare an attribute to an expression based on attributes.
+
+    The {comparison_op} operator can be 'equal to', 'not equal to', 'greater than', 'less than', 'greater than or equal to', and 'less than or equal to'.
+
+    The {expression} should be composed by attribute values, and use the following operators:
+    + : addition;
+    - : subtraction;
+    * : multiplication;
+    / : division;
+    % : modulus;
+    ** : exponentiation.
+    """
+
+    operators = {
+        '+' : operator.add,
+        '-' : operator.sub,
+        '*' : operator.mul,
+        '/' : operator.truediv,
+        '%' : operator.mod,
+        '**' : operator.pow,
+        'equal to' : operator.eq,
+        'not equal to' : operator.ne,
+        'greater than' : operator.gt,
+        'less than' : operator.gt,
+        'greater than or equal to' : operator.ge,
+        'less than or equal to' : operator.le,
+    }
+
+    if expression is not None:
```

Comment:
Can probably be simply
```suggestion
    if expression:
```

Reply:
Ref: [Zen of Python](https://en.wikipedia.org/wiki/Zen_of_Python) - 

> Explicit is better than implicit

I say let it stand.